assets/js/util/test/numberFormat.js

```diff
@@ -115,5 +116,98 @@ describe( 'numberFormat', () => {
 	it.each( siteKitLocales )( 'formats numbers correctly with locale variant %s', ( locale, value, expected ) => {
 		setupGoogleSiteKit( locale );
 		expect( numberFormat( value ) ).toStrictEqual( expected );
+		expect( console ).not.toHaveWarned();
+	} );
+
+	const NumberFormat = Intl.NumberFormat;
+
+	/*
+	 * This function mocks the implementation of certain errors identified in
+	 * https://github.com/google/site-kit-wp/issues/3255
+	 * so that we can test that we've properly handled them.
+	 */
+	const mockNumberFormat = () => {
+		global.Intl.NumberFormat = ( locales, options ) => {
+			if ( undefined === options ) {
+				return NumberFormat( locales, options );
+			}
+			const optionsThatError = {
+				currencyDisplay: 'narrow',
+				currencySign: 'accounting',
+				style: 'unit',
+				signDisplay: '*', // * denotes any value.
+				compactDisplay: '*',
+			};
+			for ( const [ key, value ] of Object.entries( options ) ) {
+				if ( optionsThatError[ key ] && [ value, '*' ].includes( optionsThatError[ key ] ) ) {
+					throw new Error( 'Implementation error' );
+				}
+			}
+
+			return NumberFormat( locales, options );
+		};
+	};
+
+	it( 'degrades gracefully when signDisplay is encountered in certain scenarios', () => {
```

Comment:
Let's avoid ambiguity in the test names as much as we can. In this case "certain scenarios" does not add much useful context for the reader so one needs to read the whole test to know what these are. Instead, let's use the title to describe the situation(s) where the behavior does change. For this one in particular, it would be any value other than the default.
```suggestion
	it( 'degrades gracefully when `signDisplay` has any value other than the default of `auto`', () => {
```